examples/map/masking_hmi.py

```diff
@@ -0,0 +1,85 @@
+"""
+=========================================
+Masking HMI based on the intensity of AIA
+=========================================
+
+In this example we will demonstrate how to mask out regions within
+a HMI image based on the intensity values of AIA.
+"""
+# sphinx_gallery_thumbnail_number = 3
+
+import matplotlib.pyplot as plt
+import numpy as np
+from skimage.measure import label, regionprops
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE, HMI_LOS_IMAGE
+
+###############################################################################
+# We will use an AIA 171 image from the sample data and crop it to capture a region of interest.
+
+aia = sunpy.map.Map(AIA_171_IMAGE)
+aia = aia.submap(
+    bottom_left=SkyCoord(-250, 200, unit=u.arcsec, frame=aia.coordinate_frame),
+    width=500 * u.arcsec,
+    height=400 * u.arcsec,
+)
+
+###############################################################################
+# Next we call the :meth:`~sunpy.map.GenericMap.reproject_to` to reproject the HMI Map
+# to have exactly the same grid as the AIA Map.
+# :ref:`sphx_glr_generated_gallery_map_transformations_reprojection_align_aia_hmi.py` provides more reference.
+
+hmi = sunpy.map.Map(HMI_LOS_IMAGE)
+hmi = hmi.reproject_to(aia.wcs)
+
+###############################################################################
+# Now let's plot the reprojected HMI Map.
+
+fig = plt.figure()
+ax = fig.add_subplot(projection=hmi)
+hmi.plot(axes=ax, cmap="hmimag")
```

Comment:
```suggestion
hmi.nickname = 'HMI magnetogram'
```
Move this plot to later so that we can overplot the regions on it